### PR TITLE
Add discovery session

### PR DESCRIPTION
**Pull request by dati91**
After this patch, we will be able to find nearby devices.

### CODE REVIEW

**Comment by zakorgy** [comment]
lgtm

**Comment by dati91** [comment]
@mmatyas r?

File: components/net/bluetooth_thread.rs [hunk 1]
```diff
@@ -355,30 +358,45 @@ impl BluetoothManager {
             Some(a) => a,
             None => return drop(sender.send(Err(String::from(ADAPTER_ERROR)))),
         };
-        let devices = self.get_and_cache_devices(&mut adapter);
-        if devices.is_empty() {
-            return drop(sender.send(Err(String::from(DEVICE_ERROR))));
+        let mut session_time = 5;
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:361]: what's this magic number here

File: components/net/bluetooth_thread.rs [hunk 2]
```diff
@@ -355,30 +358,45 @@ impl BluetoothManager {
...
+        let session = BluetoothDiscoverySession::create_session(adapter.get_object_path()).ok();
+
+        if let Some(ref s) = session {
+            let _ = s.start_discovery();
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:365]: I'd still prefer `unwrap()`, but that's just my opinion

File: components/net/bluetooth_thread.rs [hunk 3]
```diff
@@ -355,30 +358,45 @@ impl BluetoothManager {
             Some(a) => a,
             None => return drop(sender.send(Err(String::from(ADAPTER_ERROR)))),
         };
-        let devices = self.get_and_cache_devices(&mut adapter);
-        if devices.is_empty() {
-            return drop(sender.send(Err(String::from(DEVICE_ERROR))));
+        let mut session_time = 5;
+        let session = BluetoothDiscoverySession::create_session(adapter.get_object_path()).ok();
+
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:363]: no need for empty line

File: components/net/bluetooth_thread.rs [hunk 4]
```diff
@@ -355,30 +358,45 @@ impl BluetoothManager {
...
-                                     product_id: device.get_product_id().ok(),
-                                     product_version: device.get_device_id().ok(),
-                                     appearance: device.get_appearance().ok(),
-                                     tx_power: device.get_tx_power().ok().map(|p| p as i8),
-                                     rssi: device.get_rssi().ok().map(|p| p as i8),
-                                 });
-                return drop(sender.send(message));
+        for _ in 0..session_time {
+            let devices = self.get_and_cache_devices(&mut adapter);
+
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:372]: no need for empty line

File: components/net/bluetooth_thread.rs [hunk 5]
```diff
@@ -355,30 +358,45 @@ impl BluetoothManager {
...
-                                     vendor_id: device.get_vendor_id().ok(),
-                                     product_id: device.get_product_id().ok(),
-                                     product_version: device.get_device_id().ok(),
-                                     appearance: device.get_appearance().ok(),
-                                     tx_power: device.get_tx_power().ok().map(|p| p as i8),
-                                     rssi: device.get_rssi().ok().map(|p| p as i8),
-                                 });
-                return drop(sender.send(message));
+        for _ in 0..session_time {
+            let devices = self.get_and_cache_devices(&mut adapter);
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:371]: maybe `possible_devices`? just an idea

File: components/net/bluetooth_thread.rs [hunk 6]
```diff
@@ -355,30 +358,45 @@ impl BluetoothManager {
...
-                                     vendor_id_source: device.get_vendor_id_source().ok(),
-                                     vendor_id: device.get_vendor_id().ok(),
-                                     product_id: device.get_product_id().ok(),
-                                     product_version: device.get_device_id().ok(),
-                                     appearance: device.get_appearance().ok(),
-                                     tx_power: device.get_tx_power().ok().map(|p| p as i8),
-                                     rssi: device.get_rssi().ok().map(|p| p as i8),
-                                 });
-                return drop(sender.send(message));
+        for _ in 0..session_time {
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:370]: I'm not sure `for` is the best here if you always want to run the body at least once.

File: components/net/bluetooth_thread.rs [hunk 7]
```diff
@@ -16,6 +17,8 @@ use net_traits::bluetooth_thread::{BluetoothResult, BluetoothServiceMsg, Bluetoo
 use std::borrow::ToOwned;
 use std::collections::HashMap;
 use std::string::String;
+use std::thread;
```
Comment (mmatyas) [inline components/net/bluetooth_thread.rs:20]: what's the difference between `util::thread` and `std::thread`?

**Pull request closed without merging**